src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1516,6 +1516,14 @@ qlonglong TorrentImpl::nextAnnounce() const
     return lt::total_seconds(m_nativeStatus.next_announce);
 }
 
+static const qreal MONTH_SECONDS = 2629746;  // 1 month, as in C++20
```

Comment:
How did this number came from?
```
28 days: 60*60*24*28 = 2419200 sec
30 days: 60*60*24*30 = 2592000 sec
31 days: 60*60*24*31 = 2678400 sec
```

Reply:
Hello and thank you for reviewing this PR, @Chocobo1! 🙇🏼

<details>
<summary>Where it came from 🔍🕵🏽</summary>

`2629746` number came from [the C++20 standard](https://isocpp.org/std/the-standard). Sadly, the standard itself isn't available for free, but [cppreference.com mentions](https://en.cppreference.com/w/Cppreference:FAQ#Which_revision_of_the_C.2B.2B_Standard_does_this_reference_adhere_to.3F) that the final draft of C++20 is [N4860](https://wg21.link/n4860). On page 1227 of that draft you can find these definitions:
```c++
using days = duration<signed integer type of at least 25 bits ,
ratio_multiply<ratio<24>, hours::period>>;
using weeks = duration<signed integer type of at least 22 bits ,
ratio_multiply<ratio<7>, days::period>>;
using years = duration<signed integer type of at least 17 bits ,
ratio_multiply<ratio<146097, 400>, days::period>>;
using months = duration<signed integer type of at least 20 bits ,
ratio_divide<years::period, ratio<12>>>;
```

[LLVM](https://llvm.org) [implements](https://github.com/llvm/llvm-project/blob/llvmorg-17.0.6/libcxx/include/__chrono/duration.h#L312-L315) those durations this way:
```c++
typedef duration<     int, ratio_multiply<ratio<24>, hours::period>>         days;
typedef duration<     int, ratio_multiply<ratio<7>,   days::period>>         weeks;
typedef duration<     int, ratio_multiply<ratio<146097, 400>, days::period>> years;
typedef duration<     int, ratio_divide<years::period, ratio<12>>>           months;
```

[GCC](https://gcc.gnu.org) [implements](https://github.com/gcc-mirror/gcc/blob/releases/gcc-13.2.0/libstdc%2B%2B-v3/include/bits/chrono.h#L927) `months` this way:
```c++
    using months	= duration<_GLIBCXX_CHRONO_INT64_T, ratio<2629746>>;
```

Also, there are some [insightful notes on cppreference.com](https://en.cppreference.com/w/cpp/chrono/duration#Helper_types):
> Each of the predefined duration types days, weeks, months and years covers a range of at least ±40000 years. years is equal to 365.2425 days (the average length of a Gregorian year). months is equal to 30.436875 days (exactly 1/12 of years).
</details>

Because of [this line in the Changelog](https://github.com/qbittorrent/qBittorrent/blob/master/Changelog#L259) and the lack of further C++ mentions there, I was thinking that qBittorrent supports C++17 🤦🏼‍♂️

But finally, while preparing this answer, I found that qBittorrent is already switched to C++20 in #19336, so I'm going to update the PR using `std::chrono::months` directly.

Reply:
I've updated my changes.
The `MONTH_SECONDS` constant has been removed in favor of the custom `months` duration that reuses the period from `std::chrono::months`. That custom duration is used to produce floating-point `activeMonths` numbers.